### PR TITLE
spread2 fix: add maxSize  when missing from `start`

### PR DESCRIPTION
**Pull request by CeresBarros**
If `start` comes from a previous `spread2` call to which `maxSize` was not passed, and `maxSize` if passed as an argument, a `maxSize` column needs to be added to `clusterDT` (coming from `attr(start, 'spreadState')`.

* I proposed adding this column (this requires running `as.data.table` on `attr(start, 'spreadState')`)

TO DO:
* for some reason, `maxSize` is not entirely respected when using this new  mechanism and final fire sizes sometimes exceed the maximum by a few pixels.

Additional changes:
* a minor change in a message that needs to be checked

### CODE REVIEW

**Comment by CeresBarros** [comment]
@eliotmcintire you're probably the best person to check this PR.

File: R/spread2.R [hunk 1]
```diff
@@ -515,7 +524,7 @@ spread2 <- function(landscape, start = ncell(landscape) / 2 - ncol(landscape) /
           sizeType <- if (!anyNA(exactSize)) "exactSize" else "maxSize"
           message(
             sizeType, " provided. ",
-            "It does not match with size attr(start, 'cluster')$maxSize. ",
+            "It does not match with size attr(start, 'spreadState')$maxSize. ",
```
Comment (CeresBarros) [inline R/spread2.R:527]: @eliotmcintire can you please check this?

File: R/spread2.R [hunk 2]
```diff
@@ -506,7 +506,16 @@ spread2 <- function(landscape, start = ncell(landscape) / 2 - ncol(landscape) /
...
     if (!is.null(attr(start, "spreadState"))) {
-      clusterDT <- attr(start, "spreadState")$clusterDT
+      ## as.data.table is necessary to use `set` to add new columns.
+      clusterDT <- as.data.table(attr(start, "spreadState")$clusterDT)
+
+      ## make sure maxSize column exists when maxSize argument is passed.
+      if (!anyNA(maxSize) & is.null(clusterDT$maxSize)) {
+        message("maxSize provided, but not present in attr(start, 'spreadState')$maxSize. ",
+                "Using the maxSize provided: ", maxSize)
+        set(clusterDT, NULL, "maxSize", maxSize)
```
Comment (CeresBarros) [inline R/spread2.R:516]: @eliotmcintire, note that this is still allowing final fire sizes to exceed the maximum by a few pixels, which does not seem to happen when `maxSize` is defined in the first `spread2` call.

**Comment by achubaty** [comment]
@CeresBarros is this now ready for review?

**Comment by CeresBarros** [comment]
yes

**Pull request merged**